testsuite/forge/src/interface/node.rs

```diff
@@ -69,7 +62,100 @@ pub trait Node {
 }
 
 /// Trait used to represent a running Validator
-pub trait Validator: Node {}
+pub trait Validator: Node {
+    fn check_connectivity(&self, expected_peers: usize) -> Result<bool> {
+        if expected_peers == 0 {
+            return Ok(true);
+        }
+
+        self.get_connected_peers(NetworkId::Validator, None)
+            .map(|maybe_n| maybe_n.map(|n| n >= expected_peers as i64).unwrap_or(false))
+    }
+}
 
 /// Trait used to represent a running FullNode
 pub trait FullNode: Node {}
+
+impl<T: ?Sized> NodeExt for T where T: Node {}
```

Comment:
What's the reason for adding an Extension trait here if we could just put it directly in Node?  My understanding was extension traits were for adding functionality outside of the original crate.

Reply:
It makes it impossible for implementers to override the default implementation. It also makes things a bit more readable.